src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,93 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int THIS_IS_ZERO = 0;
+    private static final int THIS_IS_ONE = 1;
+    private static final int THIS_IS_TWO = 2;
+    private T[] currentArray = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        checkGrow();
+        currentArray[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
+        chekIndex(index);
+        checkGrow();
+        System.arraycopy(currentArray, index, currentArray, index + THIS_IS_ONE, size - index);
+        currentArray[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = THIS_IS_ZERO; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        chekIndex(index);
+        return currentArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        chekIndex(index);
+        currentArray[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        chekIndex(index);
+        T removedData = currentArray[index];
+        System.arraycopy(currentArray, index + THIS_IS_ONE, currentArray,
+                index, size - THIS_IS_ONE - index);
+        size--;
+        return removedData;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T removedElement;
+        for (int i = THIS_IS_ZERO; i < size; i++) {
+            if (element == currentArray[i] || element != null && element.equals(currentArray[i])) {
+                removedElement = currentArray[i];
+                remove(i);
+                return removedElement;
+            }
+        }
+        throw new NoSuchElementException("Element not exist");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == THIS_IS_ZERO;
+    }
+
+    private void checkGrow() {
+        if (size == currentArray.length) {
+            currentArray = Arrays.copyOf(currentArray, currentArray.length
+                    + currentArray.length / THIS_IS_TWO);
```

Comment:
Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements.